pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -666,14 +692,25 @@ func convertResolution(resolution model.Resolution) apiv2.Cluster_DiscoveryType
 }
 
 // conditionallyConvertToIstioMtls fills key cert fields for all TLSSettings when the mode is `ISTIO_MUTUAL`.
+// If the (input) TLS setting is nil (i.e not set), *and* the service mTLS mode is STRICT, it also
+// creates and populates the config as if they are set as ISTIO_MUTUAL.
 func conditionallyConvertToIstioMtls(
 	tls *networking.TLSSettings,
 	serviceAccounts []string,
 	sni string,
 	proxy *model.Proxy,
+	serviceMTLSMode authn_v1alpha1_applier.MutualTLSMode,
 ) *networking.TLSSettings {
 	if tls == nil {
-		return nil
+		if serviceMTLSMode != authn_v1alpha1_applier.MTLSStrict {
+			// Destination service is not in strict-mTLS mode, do nothing.
+			return nil
+		}
+
+		// Always use ISTIO_MUTUAL if destination service is strict-mTLS
+		tls = &networking.TLSSettings{
```

Comment:
nit. is it possible to have a global constant here? since this is not mutated at all?

Reply:
Actually it can, as the purpose of the function.